### PR TITLE
Add new API & update search payload for Purview Catalog

### PR DESCRIPTION
**Pull request by yifan-zhou922**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by yifan-zhou922** [comment]
Hi @ruowan @raych1, I want to update some examples for Purview Catalog. Could you please help me review this PR? I have no clues why the model validation and breaking change validation fail. Thanks.

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json [hunk 1]
```diff
@@ -6000,13 +6065,17 @@
         "@search.highlights": {
           "$ref": "#/definitions/SearchHighlights"
         },
-        "@search.text": {
```
Comment (raych1) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json:6003]: @search.text is removed here but it's still referenced in the example files which caused the model validation errors. 

in operationId "Discovery_Query", "@search.text" is an additional property, so model validation report OBJECT_ADDITIONAL_PROPERTIES error

Comment (yifan-zhou922) [reply specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json:6003]: Thanks @raych1. I have resolved this error. Could you please help me check whether I should resolve the other three model validation errors which I think are not caused by this PR change. Many thanks.

**Comment by raych1** [comment]
@jianyexi , can you have a look at `breaking change` and `lintDiff` exceptions?

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/examples/Glossary_GetImportCSVOperationStatus.json [hunk 2]
```diff
@@ -7,6 +7,9 @@
   },
   "responses": {
     "202": {
+      "headers": {
+        "location": "/api/glossary/terms/import/3ffca165-75a7-4987-ba71-cfc0a273d6b7"
```
Comment (jianyexi) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/examples/Glossary_GetImportCSVOperationStatus.json:11]: the location should match  "^(http|https)://.*",

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/examples/Glossary_GetImportCSVOperationStatusByGloassaryName.json [hunk 3]
```diff
@@ -8,6 +8,9 @@
   },
   "responses": {
     "202": {
+      "headers": {
+        "location": "/api/glossary/terms/import/99e60bea-6f70-4fcb-966a-a313bb2738ef"
```
Comment (jianyexi) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/examples/Glossary_GetImportCSVOperationStatusByGloassaryName.json:12]: the location should match "^(http|https)://.*",

**Comment by jianyexi** [comment]
> @jianyexi Jianye Xi FTE , can you have a look at `breaking change` and `lintDiff` exceptions?

Seems schema validator will check the "location" in the example should match  "^(http|https)://.*", just fix the example I commented can resolve the exception

**Comment by raych1** [comment]
@jhendrixMSFT , I just realized this PR is data plane PR and I was assigned incorrectly.  Can you please review this PR and follow up on it?

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json [hunk 4]
```diff
@@ -2468,10 +2473,8 @@
             "$ref": "./examples/Discovery_Query_AssetType.json"
           },
           "Discovery_Query_Term": {
-            "$ref": "./examples/Discovery_Query_Term.json"
-          },
-          "Discovery_Query_TermOfGlossary": {
-            "$ref": "./examples/Discovery_Query_TermOfGlossary.json"
+            "$ref": "./examples/Discovery_Query_TermAssignment.json",
+            "description": "Find assets by the term assigned."
```
Comment (jhendrixMSFT) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json:2477]: This `description` element is invalid and I believe is why the breaking changes tool is failing.  Can you please remove it?

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json (hunk 4 above)
Comment (yifan-zhou922) [reply specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2021-05-01-preview/purviewcatalog.json]: Thanks. I have removed this line and will let arch aboard review and approve the breaking changes first.

Another question, do you know whether adding descriptions to each example is supported? If yes, how can I add this to the swagger file. If not, will we support this in the future? Providing example descriptions can make customer easier to understand which example can be referred to under different situations. Many thanks.

**Pull request closed without merging**